ansible/roles/telegraf/tasks/main.yml

```diff
@@ -0,0 +1,80 @@
+---
+# Copyright (C) 2021 Dr. Serge Victor <https://dr.sergevictor.eu/>
+# Copyright (C) 2020 DebOps <https://debops.org/>
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Import custom Ansible plugins
+  import_role:
+    name: 'ansible_plugins'
+
+- name: Import DebOps global handlers
+  import_role:
+    name: 'global_handlers'
+
+- name: Divert the original telegraf main configuration file
+  dpkg_divert:
+    path: '/etc/telegraf/telegraf.conf'
+    state: 'present'
+    delete: true
+  notify: [ 'Restart Telegraf' ]
+
+- name: Configure Telegraf instance leading file
+  template:
+    src: 'etc/telegraf/telegraf.conf.j2'
+    dest: '/etc/telegraf/telegraf.conf'
+    owner: 'root'
+    group: 'root'
+    mode: '0644'
+  notify: [ 'Restart Telegraf' ]
+
+- name: Remove plugins configuration if requested
+  file:
+    path: '/etc/telegraf/telegraf.d/{{ item.name }}.conf'
+    state: 'absent'
+  loop: '{{ telegraf__configuration_combined | parse_kv_items }}'
+  when: item.state|d('present') == 'absent' and item.config|d()
+  notify: [ 'Restart Telegraf' ]
+
+- name: Find plugins which are in filesystem
+  find:
+    paths: /etc/telegraf/telegraf.d
+    patterns: '*.conf'
+  register: telegraf_existing_plugins_filenames
+
+- name: Backup and disactivate plugins unexisting in desired configuration
+  command:
+    cmd: mv -f '{{ item.path | quote }}' '{{ item.path | quote }}.inactive'
+    removes: item.path
+  loop: '{{ telegraf_existing_plugins_filenames.files }}'
+  when: item.path | basename | splitext | first not in telegraf__configuration_combined | parse_kv_items | map(attribute="name")
+  notify: [ 'Restart Telegraf' ]
```

Comment:
Are you moving around files packaged in a `.deb` package? This will break things when the `telegraf` package is upgraded. Is that really needed? Wait, no, you are moving around configuration files... That's even worse, are these configuration files included with the package? Run `dpkg-query -s telegraf` and check if these are marked as Conffiles. If yes, then you should divert them instead so that `dpkg` can properly track them.